app/src/main/java/com/example/Egida/presentation/viewModel/SettingViewModel.kt

```diff
@@ -0,0 +1,45 @@
+package com.example.Egida.presentation.viewModel
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.Egida.Dependencies
+import com.example.Egida.domain.entity.UserDB
+import com.example.Egida.domain.useCase.UserDBUseCase
+import com.example.Egida.domain.useCase.userAUTH.UserAUTHUseCase
+import com.example.Egida.utils.USER_DB
+
+class SettingViewModel : ViewModel() {
+
+
+    private val userAUTHUseCase: UserAUTHUseCase by lazy { Dependencies.userAUTHUseCase() }
+    private val userDBUseCase: UserDBUseCase by lazy { Dependencies.userDBUseCase() }
+    var firstName: String = ""
+    var lastName: String = ""
+    var checkAgreement: Boolean = false
+    var phoneNumber: String = ""
+    var photoURL: String = ""
+    var height: Int = 180
+    var weight: Int = 80
```

Comment:
Magic constants again. And all other fields initialised with an empty. Two different approaches.